docs/advanced_topics.rst

```diff
@@ -0,0 +1,43 @@
+Advanced topics
+=================
+
+In this section, you can find some advanced functionalities and topics that are just partially mentioned in the API description. For the typical use case, DALI default configuration should perform well out of the box and you don't need to worry about those.
+
+DALI internal data format
+-------------------------
+
+Internally, DALI processes data in NHWC format. To reshape it to NCHW please use ``\*Permute`` operator family. It also means that if mentioned operators are used in the middle of the pipeline then next operations will take the wrong assumption about processed data format and the result is undefined.
+
+Thread affinity
+---------------
+
+The rationale behind this functionality it to allow pining DALI threads to given CPU. It can save the overhead of worker threads jumping from core to core and increase performance for CPU heave workloads.
+It is possible to set CPU thread affinity for DALI using ``DALI_AFFINITY_MASK`` environment variable. ``DALI_AFFINITY_MASK`` should be a comma separated list of CPU ids numbers that will be assigned to corresponding DALI threads. A number of DALI threads is set during pipeline construction by ``num_threads`` argument, ``set_affinity`` enables setting thread affinity for the CPU worker threads. It is worth to mention that due to performance reason nvJPEG decoder operator houses threads on its own and they are always affined.
+When ``DALI_AFFINITY_MASK`` is empty affinity is set based on the result returned by nvmlDeviceGetCpuAffinity.
```

Comment:
It is possible to set the CPU thread affinity for DALI using the ``DALI_AFFINITY_MASK`` environment variable. ``DALI_AFFINITY_MASK`` should be a comma separated list of CPU ids numbers which will be assigned to corresponding DALI threads. 

The number of DALI threads are set during the pipeline construction by ``num_threads`` argument, and the ``set_affinity`` enables setting the thread affinity for the CPU worker threads. Note that for performance reasons the nvJPEG decoder operator houses threads on its own, and they are always affined.

Reply:
Adjusted